### ISSUE TITLE
Multiple Container Instances

### ISSUE THREAD

**Issue by daniel-g-wood** [zenstruck/foundry#225]
I'm not sure that this is an issue exactly, but I'm working on a legacy application, and have ended up in a situation where I have multiple container instances, and therefore two instances of the Doctrine EntityManager.

I need to inject the EntityManager into one of my factories so that I can remove the auto-generated id metadata:

```
    public function __construct(EntityManagerInterface $entityManager)
    {
        parent::__construct();
        $metadata = $entityManager->getClassMetaData(self::getClass());
        $metadata->setIdGeneratorType(ClassMetadataInfo::GENERATOR_TYPE_NONE);
        $metadata->setIdGenerator(new AssignedGenerator());
    }

```

Unfortunately, this is only removing the auto-generation on the instance of the class using the injected EntityManager, but the model doing the insert comes from a different instance of the EntityManager (I used `spl_object_id()` to verify that the instances were different). This only happens when I use the `Factories` trait. Once I remove it, then the instances are the same and the tests pass, but I'm unsure what other impacts this will have.

My factories are loading from Tests\Factory, so I need to manually register and tag them as services. I'm not sure whether this is effecting it as well.

My environment:
PHPUnit 6.5.14
Symfony 4.4.13
zenstruck/foundry 1.13.2

As I said above, this may not be an issue with zenstruck/foundry. It might just be a quirk of our setup.

Loving the package by the way; it really helps to make all my tests clean and easy to understand!

**Comment by kbond** [comment]
Sounds like it could be a quirk with your setup as you say. Do you have multiple kernels then?

While recommended, the `Factories` trait isn't strictly required. These are the reasons it is required:
- when the bundle is not enabled
- when you have test-specific configuration
- for tests that don't extend `KernelTestCase`
- it solves an issue if your tests reboot the kernel - it provides foundry a lazy manager registry that is always available with the current kernel (I suspect this is causing your issue)

> My factories are loading from Tests\Factory, so I need to manually register and tag them as services. I'm not sure whether this is effecting it as well.

I don't believe this is the issue.

**Comment by daniel-g-wood** [comment]
Thanks for responding so quickly.

> Do you have multiple kernels then?

Unfortunately no, I checked for that. Just the one kernel (`App\Kernel`).

Based on the above, it looks like I don't need to include the `Factories` trait for now.

Closing the issue as it seems to be working on another of our legacy apps with a less-complex setup. So my guess there is still something weird about the way the tests are running in this one.

Thanks for your help!
